lumen/ai/agents.py

```diff
@@ -542,20 +545,24 @@ async def _create_valid_sql(self, messages, system, sources, errors=None):
         if not sql_query:
             raise ValueError("No SQL query was generated.")
 
+        sources = set(tables_to_source.values())
         if len(sources) > 1:
             mirrors = {}
-            for a_source, a_table in sources.values():
+            for a_table, a_source in tables_to_source.items():
```

Comment:
Basically reversed it and dropped one layer; I don't think the keys initially were useful.